modules/refine-local-runner/src/main/java/org/openrefine/runners/local/pll/PLL.java

```diff
@@ -94,17 +96,30 @@ public int numPartitions() {
 
     /**
      * Iterate over the elements of the given partition. If the contents of this PLL have been cached, this will iterate
-     * from the cache instead.
-     * 
+     * from the cache instead. Iteration is done with the default {@link IterationContext}.
+     *
      * @param partition
      *            the partition to iterate over
-     * @return
      */
     public CloseableIterator<T> iterate(Partition partition) {
+        return iterate(partition, IterationContext.DEFAULT);
+    }
+
+    /**
+     * Iterate over the elements of the given partition. If the contents of this PLL have been cached, this will iterate
+     * from the cache instead.
+     *
+     * @param partition
+     *            the partition to iterate over
+     * @param context
+     *            some additional information to configure the iteration, relaxing the completeness requirements of the
+     *            generated data
+     */
+    public CloseableIterator<T> iterate(Partition partition, IterationContext context) {
```

Comment:
This changes the PLL interface to let the user pick between sychronous and asynchronous iteration. To avoid such interface changes, I have introduced `IterationContext` which could potentially hold more parameters in the future, without changing the method signature again. One other parameter I imagine introducing is a selection of columns to fetch, to enable columnar reading of the project data, assuming the underlying serialization format supports it.